src/arch/riscv/traps.S

```diff
@@ -97,12 +97,9 @@ trap_entry:
   STORE   x1, (33*REGBYTES)(t0)
 
   /* Check if it's an interrupt */
-  srli s2, s0, (CONFIG_WORD_SIZE - 1)
-  li   s1, 0x1
-  beq  s2, s1, interrupt
+  bltz s0, interrupt
```

Comment:
nice

Reply:
This could be a separate commit? My read of the commit message is about the code change below.